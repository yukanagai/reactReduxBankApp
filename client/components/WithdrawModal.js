import React from 'react';
import ModalTemplate from './ModalTemplate.js';

class WithdrawPage extends React.Component {

	constructor(props) {
		super(props);
	}

	handleSubmit(val) {
		if (val) {
			this.props.withdraw(val);
		}
	}

	handleCancel() {
		this.props.hideModal();
	}

	render() {
		return (
			<ModalTemplate
				headerText='How much would you like to withdraw?' 
				handleLeftButtonAction={this.handleCancel.bind(this)}
				handleRightButtonAction={this.handleSubmit.bind(this)}
				{...this.props} />
		);
	}

}

export default WithdrawPage;