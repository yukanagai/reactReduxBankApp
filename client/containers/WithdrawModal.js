import React from 'react';
import ModalTemplate from '../components/ModalTemplate.js';

class WithdrawPage extends React.Component {
	constructor(props) {
		super(props);
	}
	
	onCancelAction() {
		// DISPATCHING TOGGLE MODAL ACTION TO HIDE MODAL
		this.props.toggleModal();
	}

	onWithdrawAction(val) {
		// DISPATCHING WITHDRAW ACTION
		this.props.withdraw(val);
	}

	render() {
		return (
			<ModalTemplate
				headerText='How much would you like to withdraw?' 
				handleLeftButtonAction={this.onCancelAction.bind(this)}
				handleRightButtonAction={this.onWithdrawAction.bind(this)}
				{...this.props}
			/>
		);
	}
}

export default WithdrawPage;