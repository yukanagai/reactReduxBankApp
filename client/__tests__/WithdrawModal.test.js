import React from 'react';
import renderer from 'react-test-renderer';
import WithdrawModal from '../containers/WithdrawModal.js';

// Snapshot for WithdrawModal React Component
test('WITHDRAW MODAL --- Snapshot',()=>{
    const tree = renderer.create(
    	<WithdrawModal headerText='Header text test' />
    ).toJSON();
   	expect(tree).toMatchSnapshot();
});