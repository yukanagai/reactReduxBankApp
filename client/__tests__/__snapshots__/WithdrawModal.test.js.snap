// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WITHDRAW MODAL --- Snapshot 1`] = `
<div
  className="home"
>
  <div
    className="content"
  >
    <h3>
      Header text test
    </h3>
    <input
      className="ui input"
      maxLength="10"
      placeholder="numbers only"
      type="number"
    />
  </div>
  <div
    className="buttonSection"
  >
    <a
      onClick={[Function]}
      style={Object {}}
    >
      <button
        className="ui button"
        id="cancel"
        onClick={[Function]}
      >
        Cancel
      </button>
    </a>
    <a
      onClick={[Function]}
      style={Object {}}
    >
      <button
        className="ui button"
        id="submit"
        onClick={[Function]}
      >
        Submit
      </button>
    </a>
  </div>
</div>
`;
